core/templates/domain/email-dashboard/email-dashboard-query-object.factory.ts

```diff
@@ -34,13 +34,13 @@ export interface EmailDashboardQueryBackendDict {
 export class EmailDashboardQuery {
   id: string;
   status: string;
-  numQualifiedUsers: number;
-  submitterUsername: string;
-  createdOn: string;
+  numQualifiedUsers?: number;
+  submitterUsername?: string;
+  createdOn?: string;
```

Comment:
In what cases can these be `null` or `undefined`?

Reply:
The ```EmailDashboardQueryDict``` interface had these types as ```string | undefined``` or ```number | undefined```. So, instead of changing the interface, I decided to change the type of these wherever they were referenced. 

After going through ```../domain/email-dashboard/``` and ```../pages/email-dashboard-pages``` I reckon these would always have some string/number value.  

I'll change this.

Reply:
But, if you see line 100 of ```email-dashboard-data-service.spec.ts```
```       
        var queryData = emailDashboardQueryObjectFactory.createFromQueryDict({
          id: 'qnew',
          status: 'processing'
        });
```

```numQualifiedUsers```, ```submitterUsername``` and ```createdOn``` are undefined.

Reply:
If they are undefined just in the spec files (test files) then it would rather make sense to set them to some dummy value in the tests.

Reply:
What dummy values do you suggest for:
- ```numQualifiedUsers```
- ```submitterUsername```
- ```createdOn``` 

would 0 and empty strings work?

Reply:
Try it, I think it should work.

Reply:
After changing the values in test files, 2 tests are failing:

![error_tests](https://user-images.githubusercontent.com/26626415/94702110-04100c80-035b-11eb-8d4e-d721f1b14d21.png)

Reply:
@vojtechjelinek PTAL, thanks

Reply:
There is not much context I can work with, post the second error and the actual contents of the spec file. Also, check taht you didn't forgot about some place.

Reply:
@vojtechjelinek  How do I share the spec file? Can't push the changes because of the errors.

These 2 tests are causing errors:
https://github.com/oppia/oppia/blob/a3f7191fe7a9db9ec32bc2bf1de044a8943fef57/core/templates/pages/email-dashboard-pages/email-dashboard-data.service.spec.ts#L55-L89

https://github.com/oppia/oppia/blob/a3f7191fe7a9db9ec32bc2bf1de044a8943fef57/core/templates/pages/email-dashboard-pages/email-dashboard-data.service.spec.ts#L90-L128

Reply:
You can push with `--no-verify` flag so I can see the code.

Reply:
Did it, PTAL

Reply:
Took a look and cannot find anything just from looking at it. Please try to add some debug logs to see what are the values so that you can evaluate the code.

Reply:
These are the errors while running the tests. 

![Screenshot from 2020-10-06 07-55-50](https://user-images.githubusercontent.com/26626415/95151878-65940900-07a9-11eb-9ebb-fd7f9b3aab00.png)

Other than this, how should I get the logs?

Reply:
I mean you should modify the code and debug the tests yourselves. I will have time to look into this more after next Monday. Maybe @nishantwrp can help.

Reply:
I've discussed this issue offline with @aishwary023.

Reply:
@vojtechjelinek  @nishantwrp  PTAL at the changes. Thanks.